pkg/distributors/test/MerkleRedeem.test.ts

```diff
@@ -0,0 +1,273 @@
+import { ethers } from 'hardhat';
+import { BytesLike, BigNumber } from 'ethers';
+import { expect } from 'chai';
+import { Contract } from 'ethers';
+
+import { deploy } from '@balancer-labs/v2-helpers/src/contract';
+import { expectBalanceChange } from '@balancer-labs/v2-helpers/src/test/tokenBalance';
+import Token from '@balancer-labs/v2-helpers/src/models/tokens/Token';
+import TokenList from '@balancer-labs/v2-helpers/src/models/tokens/TokenList';
+import { GeneralPool } from '@balancer-labs/v2-helpers/src/models/vault/pools';
+import { encodeJoin } from '@balancer-labs/v2-helpers/src/models/pools/mockPool';
+
+import { bn } from '@balancer-labs/v2-helpers/src/numbers';
+import { MAX_UINT256 } from '@balancer-labs/v2-helpers/src/constants';
+import { MerkleTree } from '../lib/merkleTree';
+
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/dist/src/signer-with-address';
+
+function encodeElement(address: string, balance: BigNumber): string {
+  return ethers.utils.solidityKeccak256(['address', 'uint'], [address, balance]);
+}
+
+describe('MerkleRedeem', () => {
+  let tokens: TokenList,
+    rewardTokens: TokenList,
+    rewardToken: Token,
+    vault: Contract,
+    authorizer: Contract,
+    merkleRedeem: Contract;
+
+  let admin: SignerWithAddress,
+    lp1: SignerWithAddress,
+    lp2: SignerWithAddress,
+    lp3: SignerWithAddress,
+    mockAssetManager: SignerWithAddress;
+  let poolId: string;
+  const rewardTokenInitialBalance = bn(100e18);
+  const tokenInitialBalance = bn(200e18);
+
+  before('deploy base contracts', async () => {
+    [admin, lp1, lp2, lp3, mockAssetManager] = await ethers.getSigners();
+  });
+
+  beforeEach('set up tokens and redeem contract', async () => {
+    tokens = await TokenList.create(['SNX', 'MKR'], { sorted: true });
+
+    rewardTokens = await TokenList.create(['DAI'], { sorted: true });
+    rewardToken = rewardTokens.DAI;
+
+    // Deploy Balancer Vault
+    authorizer = await deploy('v2-vault/Authorizer', { args: [admin.address] });
+    vault = await deploy('v2-vault/Vault', { args: [authorizer.address, tokens.SNX.address, 0, 0] });
+
+    merkleRedeem = await deploy('MerkleRedeem', {
+      args: [vault.address, rewardToken.address],
+    });
+    await rewardTokens.mint({ to: admin.address, amount: rewardTokenInitialBalance });
+    await rewardTokens.approve({ to: merkleRedeem.address, from: [admin] });
+
+    // deploy pool and add liquidity
+    const specialization = GeneralPool;
+    const pool = await deploy('v2-vault/MockPool', { args: [vault.address, specialization] });
+    poolId = await pool.getPoolId();
+
+    await tokens.mint({ to: lp1, amount: tokenInitialBalance });
+    await tokens.approve({ to: vault.address, from: [lp1] });
+
+    const assets = tokens.addresses;
+    const assetManagers = [mockAssetManager.address, mockAssetManager.address];
```

Comment:
```suggestion
    const assetManagers = [mockAssetManager.address, mockAssetManager.address];
```

I think we don't need these, I'd just pass the zero address instead.